src/modbus.c

```diff
@@ -386,7 +404,17 @@ int _modbus_receive_msg(modbus_t *ctx, uint8_t *msg, msg_type_t msg_type)
         rc = ctx->backend->select(ctx, &rset, p_tv, length_to_read);
         if (rc == -1) {
             _error_print(ctx, "select");
-            if (ctx->error_recovery & MODBUS_ERROR_RECOVERY_LINK) {
+            if (ctx->error_recovery & MODBUS_ERROR_RECOVERY_LINK &&
+                ctx->backend->backend_type == _MODBUS_BACKEND_TYPE_TCP) {
```

Comment:
Same here.

Reply:
But does MODBUS_ERROR_RECOVERY_LINK make sense for _MODBUS_BACKEND_TYPE_RTU ? that's why I have added that test.

Reply:
Wrong, `modbus_flush()` is useful in RTU and we need to handle unplug/plug of serial device too.
BTW, PR should concern only one subject.